### PR TITLE
Implement XDG Toplevel move and resize requests.

### PR DESCRIPTION
**Pull request by Leon-Plickat**
Pretty simple patch, since the code that does the actual work is already there, it just needs to be called. Using `Cursor.Mode` is a bit awkward though, since its `self` wants a `Cursor`.

### CODE REVIEW

File: river/InputManager.zig [hunk 1]
```diff
@@ -151,3 +151,14 @@ fn handleNewInput(listener: ?*c.wl_listener, data: ?*c_void) callconv(.C) void {
     // TODO: suport multiple seats
     self.defaultSeat().addDevice(device);
 }
+
+/// Returns a seat matching a wlr_seat.
+pub fn seatFromWlrSeat(self: *Self, wlr_seat: *c.wlr_seat) ?*Seat {
```
Comment (ifreund) [inline river/InputManager.zig:156]: We store a pointer to the river `Seat` in the `data` field of the `wlr_seat`, this function shouldn't be needed.

**Review by ifreund — COMMENTED** [review]
> Using Cursor.Mode is a bit awkward though, since its self wants a Cursor.

Yeah, this makes me realize that the functions should probably be members of `Cursor` not `Cursor.Mode`. Feel free to move them in a separate commit if you like.

**Comment by Leon-Plickat** [comment]
Moved the functions from `Cursor.Mode` to `Cursor`; Now they are straight forward to use.

**Review by ifreund — APPROVED** [review]
LGTM, thanks!

**Pull request merged**